### ISSUE TITLE
nyc fails to parse ES6 ava.config.js file

### ISSUE THREAD

**Issue by kevinji** [istanbuljs/nyc#1062]
## Link to bug demonstration repository.
See the [Travis build failure](https://travis-ci.org/honestbleeps/Reddit-Enhancement-Suite/jobs/516976630) in honestbleeps/Reddit-Enhancement-Suite#4841.

ava.config.js contains the following:
```js
/* @flow */

export default {
	files: [
		'**/__tests__/*.js',
	],
	require: [
		'@babel/register',
	],
};
```

## Expected Behavior
`cross-env NODE_ENV=test nyc ava` should succeed.

## Observed Behavior
`nyc` reports an error with parsing `export default`:
```
$ cross-env NODE_ENV=test nyc ava
✖ Error loading ava.config.js
/home/travis/build/honestbleeps/Reddit-Enhancement-Suite/node_modules/ava/lib/load-config.js:17
ReferenceError: default is not defined
    at loadConfig (/home/travis/build/honestbleeps/Reddit-Enhancement-Suite/node_modules/ava/lib/load-config.js:17:5)
    at Object.exports.run (/home/travis/build/honestbleeps/Reddit-Enhancement-Suite/node_modules/ava/lib/cli.js:24:10)
    at Object.<anonymous> (/home/travis/build/honestbleeps/Reddit-Enhancement-Suite/node_modules/ava/cli.js:10:23)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Module.replacementCompile (/home/travis/build/honestbleeps/Reddit-Enhancement-Suite/node_modules/nyc/node_modules/append-transform/index.js:58:13)
    at Module._compile (/home/travis/build/honestbleeps/Reddit-Enhancement-Suite/node_modules/pirates/lib/index.js:83:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at /home/travis/build/honestbleeps/Reddit-Enhancement-Suite/node_modules/nyc/node_modules/append-transform/index.js:62:4
    at newLoader (/home/travis/build/honestbleeps/Reddit-Enhancement-Suite/node_modules/pirates/lib/index.js:88:7)
    at Object.<anonymous> (/home/travis/build/honestbleeps/Reddit-Enhancement-Suite/node_modules/nyc/node_modules/append-transform/index.js:62:4)
```

## Forensic Information
Travis, node.js image. Related: #916.

**Comment by coreyfarrell** [comment]
I was able to reproduce a conflict between the way ava uses babel and `@babel/register` being loaded from outside ava.  I've opened a ticket at avajs/ava#2089 to see if they have ideas for how to make this less of an issue.

Good news I was able to work around this issue: add `"ignore": ["ava.config.js"]` to your babel configuration.  It is specifically the `@babel/plugin-transform-modules-commonjs` being run on ava.config.js from outside ava's control which causes this failure.
